tests/integration/apigateway/test_apigateway_kinesis.py

```diff
@@ -0,0 +1,143 @@
+    snapshot.add_transformer(snapshot.transform.kinesis_api())
+
+    stream_name = f"kinesis-stream-{short_uid()}"
+    function_name = f"lambda-consumer-{short_uid()}"
+    region_name = apigateway_client.meta.region_name
+
+    api_id, name, root_id = create_rest_apigw(
+        name="test-apigateway-to-kinesis",
+        description="test apigateway to kinesis",
+        endpointConfiguration={"types": ["REGIONAL"]},
+    )
+
+    resource_id, _ = create_rest_resource(
+        apigateway_client, restApiId=api_id, parentId=root_id, pathPart="test"
+    )
+
+    method, _ = create_rest_resource_method(
+        apigateway_client,
+        restApiId=api_id,
+        resourceId=resource_id,
+        httpMethod="POST",
+        authorizationType="NONE",
+    )
+
+    assume_role_arn = create_iam_role_with_policy(
+        RoleName=f"role-apigw-{short_uid()}",
+        PolicyName=f"policy-apigw-{short_uid()}",
+        RoleDefinition=APIGATEWAY_ASSUME_ROLE_POLICY,
+        PolicyDefinition=APIGATEWAY_KINESIS_POLICY,
+    )
+
+    create_rest_api_integration(
+        apigateway_client,
+        restApiId=api_id,
+        resourceId=resource_id,
+        httpMethod=method,
+        integrationHttpMethod="POST",
+        type="AWS",
+        credentials=assume_role_arn,
+        uri=f"arn:aws:apigateway:{region_name}:kinesis:action/PutRecord",
+        requestTemplates={
+            "application/json": json.dumps(
+                {
+                    "StreamName": stream_name,
+                    "Data": "$util.base64Encode($input.body)",
+                    "PartitionKey": "test",
+                }
+            )
+        },
+    )
+
+    create_rest_api_method_response(
+        apigateway_client,
+        restApiId=api_id,
+        resourceId=resource_id,
+        httpMethod="POST",
+        statusCode="200",
+    )
+
+    create_rest_api_integration_response(
+        apigateway_client,
+        restApiId=api_id,
+        resourceId=resource_id,
+        httpMethod="POST",
+        statusCode="200",
+    )
+
+    deployment_id, _ = create_rest_api_deployment(apigateway_client, restApiId=api_id)
+    stage = create_rest_api_stage(
+        apigateway_client, restApiId=api_id, stageName="dev", deploymentId=deployment_id
+    )
+
+    kinesis_create_stream(StreamName=stream_name, ShardCount=1)
+    wait_for_stream_ready(stream_name=stream_name)
+    stream_summary = kinesis_client.describe_stream_summary(StreamName=stream_name)
+    assert stream_summary["StreamDescriptionSummary"]["OpenShardCount"] == 1
+    stream_arn = kinesis_client.describe_stream(StreamName=stream_name)["StreamDescription"][
+        "StreamARN"
+    ]
+
+    create_lambda_function(
+        func_name=function_name,
+        handler_file=TEST_LAMBDA_PYTHON_ECHO,
+        runtime=LAMBDA_RUNTIME_PYTHON39,
+        role=lambda_su_role,
+    )
+
+    create_event_source_mapping_response = lambda_client.create_event_source_mapping(
```

Comment:
Curious about the event source mapping that connects the Kinesis stream to the Lambda - not sure if we need it? 

I think for the purpose of testing the integration between API GW -> Kinesis it would be sufficient to do something like:
* get a shard iterator for the `LATEST` events for the kinesis stream
* invoke the API Gateway endpoint (which should call `PutRecord` on the stream)
* call `GetRecords` on the stream with the shard iterator (which should return the record)

If we decide to leave the Lambda and event source source mapping in here - can we add an assertion at the end of the test that asserts that the Lambda has been invoked? (to ensure that the entire e2e roundtrip has succeeded)

Reply:
The intent was to have a test that does the full end-to-end building, not as much as the invoking of the lambda but I agree we might as well just poll the kinesis iterator.